File: src/main/java/Game.java
import com.google.common.collect.ImmutableList;
import java.util.List;

public record Game(String uuid,
                   Player redPlayer,
                   Player blackPlayer,
                   int gameTimer,
                   int moveTimer,
                   int increment,
                   List<Board> gameStates,
                   List<Move> moves,
                   GameResult resultRed,
                   GameResult resultBlack,
                   GameResultReason gameResultReason) {

    public Game {
        gameStates = ImmutableList.copyOf(gameStates);
        moves = ImmutableList.copyOf(moves);
    }

    public List<Board> getGameStates(){
        return ImmutableList.copyOf(this.gameStates);
    }

    public List<Move> getMoves(){
        return ImmutableList.copyOf(this.moves);
    }
}

File: src/main/java/Game.kt
data class Game(
    override val uuid : String,
    override val redPlayer: Player,
    override val blackPlayer: Player,
    override val gameTimer: Int,
    override val moveTimer: Int,
    override val increment: Int,
    override val moves: List<Move>,
    override val resultRed: GameResult,
    override val resultBlack: GameResult,
    override val gameResultReason: GameResultReason
) : IGame
{
    init{
        require(gameTimer > 0)
        require(moveTimer > 0)
        require(increment >= 0)
        // TODO: create separate TimedMove and/or TimedGame classes, untimed moves and games have no use for this field
        require(moves.all { it.thinkTime != null } || moves.all { it.thinkTime == null })
    }
}